aws/retry/middleware.go

```diff
@@ -74,6 +84,10 @@ func (r AttemptMiddleware) HandleFinalize(ctx context.Context, in smithymiddle.F
 					return out, metadata, fmt.Errorf("failed to rewind transport stream for retry, %w", err)
 				}
 			}
+
+			if r.LogAttempts {
+				logger.Logf(logging.Debug, "retrying request %s/%s, attempt %d", service, operation, attemptNum)
+			}
```

Comment:
Should there also be an log for an failed attempt with information about if it will or wont be retried?

Reply:
I think it makes sense to log whether we won't attempt to retry if we encounter an un-retryable, but otherwise I think it's sufficient to log the retries as shown. I'm cautious to make it to verbose, and am not sure if we should print out the full error message, or just report we encountered an unretryable error.